pallets/orml-currencies-allowance-extension/src/lib.rs

```diff
@@ -0,0 +1,261 @@
+		fn build(&self) {
+			for i in &self.allowed_currencies.clone() {
+				AllowedCurrencies::<T>::insert(i, ());
+			}
+		}
+	}
+
+	#[pallet::pallet]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	// The pallet's dispatchable functions.
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Added allowed currencies that possible to use chain extension
+		///
+		/// # Arguments
+		/// * `currencies` - list of currency id allowed to use in chain extension
+		#[pallet::call_index(0)]
+		#[pallet::weight(<T as Config>::WeightInfo::add_allowed_currencies())]
+		#[transactional]
+		pub fn add_allowed_currencies(
+			origin: OriginFor<T>,
+			currencies: Vec<CurrencyOf<T>>,
+		) -> DispatchResult {
+			ensure_root(origin)?;
+			for i in currencies.clone() {
+				AllowedCurrencies::<T>::insert(i, ());
+			}
+
+			Self::deposit_event(Event::AllowedCurrenciesAdded { currencies });
+			Ok(())
+		}
+
+		/// Remove allowed currencies that possible to use chain extension
+		///
+		/// # Arguments
+		/// * `currencies` - list of currency id allowed to use in chain extension
+		#[pallet::call_index(1)]
+		#[pallet::weight(<T as Config>::WeightInfo::remove_allowed_currencies())]
+		#[transactional]
+		pub fn remove_allowed_currencies(
+			origin: OriginFor<T>,
+			currencies: Vec<CurrencyOf<T>>,
+		) -> DispatchResult {
+			ensure_root(origin)?;
+			for i in currencies.clone() {
+				AllowedCurrencies::<T>::remove(i);
+			}
+
+			Self::deposit_event(Event::AllowedCurrenciesDeleted { currencies });
+			Ok(())
+		}
+	}
+}
+
+#[allow(clippy::forget_non_drop, clippy::swap_ptr_to_ref, clippy::forget_ref, clippy::forget_copy)]
+#[cfg_attr(test, mockable)]
+impl<T: Config> Pallet<T> {
+	// Check the amount approved to be spent by an owner to a delegate
+	pub fn is_allowed_currency(asset: CurrencyOf<T>) -> bool {
+		return AllowedCurrencies::<T>::get(asset) == Some(())
+	}
+
+	// Check the amount approved to be spent by an owner to a delegate
+	pub fn allowance(
+		asset: CurrencyOf<T>,
+		owner: &T::AccountId,
+		delegate: &T::AccountId,
+	) -> BalanceOf<T> {
+		Approvals::<T>::get((asset, &owner, &delegate)).unwrap_or_else(Zero::zero)
+	}
+
+	/// Creates an approval from `owner` to spend `amount` of asset `id` tokens by 'delegate'
+	/// while reserving `T::ApprovalDeposit` from owner
+	///
+	/// If an approval already exists, the new amount is added to such existing approval
+	pub fn do_approve_transfer(
+		id: CurrencyOf<T>,
+		owner: &T::AccountId,
+		delegate: &T::AccountId,
+		amount: BalanceOf<T>,
+	) -> DispatchResult {
+		ensure!(Self::is_allowed_currency(id), Error::<T>::CurrencyNotLive);
+		Approvals::<T>::try_mutate((id, &owner, &delegate), |maybe_approved| -> DispatchResult {
+			let mut approved = match maybe_approved.take() {
+				// an approval already exists and is being updated
+				Some(a) => a,
+				// a new approval is created
+				None => Default::default(),
+			};
+
+			approved = approved.saturating_add(amount);
```

Comment:
That was my comment from the other PR: I think this function should fail if there is an overflow instead of executing a saturating addition.

Reply:
@TorstenStueber 
Yes. you are right! Fixed.
then [pallet-assets](https://github.com/paritytech/substrate/blob/master/frame/assets/src/functions.rs#L818) code has this issue also.